### ISSUE TITLE
Getting an error when I try to do training.

### ISSUE THREAD

**Issue by Etienne66** [csbhr/CDVD-TSP#24]
Your pretrained models do a decent job but I was hoping to add in the REDS data to get even better results. However I can't seem to get the training to work due to the following errors. I figured out how to resolve the two warnings but not sure about the actual error. I'm using pytorch 1.81 with cuda 10.2.
```python
C:\ProgramData\Miniconda3\lib\site-packages\torch\optim\lr_scheduler.py:129: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
C:\ProgramData\Miniconda3\lib\site-packages\torch\optim\lr_scheduler.py:369: UserWarning: To get the last learning rate computed by the scheduler, please use `get_last_lr()`.
  warnings.warn("To get the last learning rate computed by the scheduler, "
Epoch   2 with Lr 1.00e-4
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\spawn.py", line 116, in spawn_main
    exitcode = _main(fd, parent_sentinel)
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\spawn.py", line 125, in _main
    prepare(preparation_data)
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\spawn.py", line 236, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\spawn.py", line 287, in _fixup_main_from_path
    main_content = runpy.run_path(main_path,
  File "C:\ProgramData\Miniconda3\lib\runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "C:\ProgramData\Miniconda3\lib\runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "C:\ProgramData\Miniconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "F:\workspaces\CDVD-TSP\code\main.py", line 21, in <module>
    t.train()
  File "F:\workspaces\CDVD-TSP\code\trainer\trainer_cdvd_tsp.py", line 34, in train
    for batch, (input, gt, _) in enumerate(self.loader_train):
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\utils\data\dataloader.py", line 355, in __iter__
    return self._get_iterator()
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\utils\data\dataloader.py", line 301, in _get_iterator
    return _MultiProcessingDataLoaderIter(self)
  File "C:\ProgramData\Miniconda3\lib\site-packages\torch\utils\data\dataloader.py", line 914, in __init__
    w.start()
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\process.py", line 121, in start
    self._popen = self._Popen(self)
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\context.py", line 224, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\context.py", line 327, in _Popen
    return Popen(process_obj)
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\popen_spawn_win32.py", line 45, in __init__
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\spawn.py", line 154, in get_preparation_data
    _check_not_importing_main()
  File "C:\ProgramData\Miniconda3\lib\multiprocessing\spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
```

**Comment by Etienne66** [comment]
I tried downgrading to Python 3.7, pytorch 0.4.1, and cuda 9.2 but I received the same error.

**Comment by csbhr** [comment]
I did not have this problem.

You can refer to this blog: https://www.freesion.com/article/740970388/
It may work.

**Comment by Etienne66** [comment]
Thanks. I had found another fork where they had done just that to the main.py. I see that you had already done it to the inference.py.

I've made several changes to support Windows 10 and still remain compatible with Linux. I also made changes to support the version 1.8.1 of PyTorch with Cuda 10.2 and Python 3.8. I put in code to make it backward compatible with PyTorch 0.4.1 with Cuda 9.2 and Python 3.7.

When I'm done I'd appreciate it if you would test it to confirm that I didn't break anything. I'm currently testing it before I upload my changes.

I've also added in logic to indicate how long it will take to complete the current epoch and the entire job.

Currently I'm running it with DeepVideoDeblurring_Dataset, GOPRO_Large, & REDS(train_blur_comp). I have 333 videos with 31,450 frames. Each epoch train seems to take around 9 hours and the testing takes another 2.5 hours on my system. I'm running a GeForce RTX 2060 SUPER with 8GB. I can only safely run a batch_size of 2. Not sure how you are getting a batch_size of 8. You must have a really good setup.
